third_party/move/move-compiler-v2/src/env_pipeline/closure_checker.rs

```diff
@@ -90,6 +130,11 @@ pub fn check_closures(env: &GlobalEnv) {
                         };
                         for captured in args {
                             let captured_ty = env.get_node_type(captured.node_id());
+                            // when capturing a value that contains option, we need to generate a warning
+                            // TODO: remove it after option type is refactored to use enum
+                            if contains_option_type(env, &captured_ty) {
+                                env.warning(&env.get_node_loc(captured.node_id()), "capturing option values is currently not supported");
```

Comment:
nit: should we mention that future release will not have this limitation?

Reply:
updated the comment